default/python/universe_generation/galaxy.py

```diff
@@ -1,35 +1,297 @@
-from random import random, uniform, randint, gauss, choice
-from math import pi, sin, cos, sqrt
+from random import random, uniform, randint, gauss
+from math import pi, sin, cos, acos, sqrt, ceil, floor
+from collections import defaultdict
 
 import freeorion as fo
 import util
 import universe_tables
 
 
-class AdjacencyGrid:
+class AdjacencyGrid(object):
     def __init__(self, universe_width):
+        assert universe_tables.MIN_SYSTEM_SEPARATION < universe_tables.MAX_STARLANE_LENGTH
         self.min_dist = universe_tables.MIN_SYSTEM_SEPARATION
-        self.cell_size = max(universe_width / 50, self.min_dist)
+        self.max_dist = universe_tables.MAX_STARLANE_LENGTH
+        self.cell_size = min(max(universe_width / 50, self.min_dist), self.max_dist/sqrt(2))
         self.width = int(universe_width / self.cell_size) + 1
-        self.grid = [[[] for _ in range(self.width)] for _ in range(self.width)]
+        self.grid = defaultdict(list)
         print "Adjacency Grid: width", self.width, ", cell size", self.cell_size
 
+    def cell(self, pos):
+        return (int(pos[0] / self.cell_size), int(pos[1] / self.cell_size))
+
     def insert_pos(self, pos):
-        self.grid[int(pos[0] / self.cell_size)][int(pos[1] / self.cell_size)].append(pos)
+        self.grid[self.cell(pos)].append(pos)
+
+    def remove(self, pos):
+        self.grid[self.cell(pos)].remove(pos)
+
+    def _square_indices_containing_cell(self, (cell_x, cell_y), radius):
+        upper_left_x = max(0, cell_x - radius)
+        upper_left_y = max(0, cell_y - radius)
+        lower_right_x = min(self.width - 1, cell_x + radius)
+        lower_right_y = min(self.width - 1, cell_y + radius)
+        return [(x, y) for x in range(upper_left_x, lower_right_x + 1)
+                for y in range(upper_left_y, lower_right_y + 1)]
+
+    def _generate_rings_around_point(self, pos, max_radius=None):
+        cell = self.cell(pos)
+        i_radius = 0
+        i_radius_max = ceil(max_radius / self.cell_size) if max_radius else (self.width + 1)
+        outer = set()
+        ring = set([1])
+        while ring and i_radius <= i_radius_max:
+            inner = outer
+            outer = set(self._square_indices_containing_cell(cell, i_radius))
+            ring = outer - inner
+            yield ring
+            i_radius += 1
+
+    # All neighbors within max_dist
+    def neighbors(self, p):
+        _neighbors = []
+        i_radius_close_enough = floor(self.max_dist / self.cell_size)
+        for i, ring in enumerate(self._generate_rings_around_point(p, self.max_dist)):
+            if i <= i_radius_close_enough:
+                _neighbors += [pos for cell in ring for pos in self.grid[cell]]
+            else:
+                _neighbors += [pos for cell in ring for pos in self.grid[cell] if
+                               util.distance(pos, p) <= self.max_dist]
+        return _neighbors
+
+    # nearest neighbor at any distance
+    def nearest_neighbor(self, p):
```

Comment:
I am happy to add the docstrings for internal functions where the docstring clarifies something about usage.

However, adding docstrings for self documenting function/variables names like `remove_pos(pos)` where the docstring will only repeat the function/variable name merely adds visual noise, in the best case.  Typically, they end up being de-coupled and becoming erroneous and noisy.  

I'd like to add doctrings, everywhere that I have a comment clarifying usage, and not add any docstrings which do not expand on the function name.